Support devkitPro pacman (fix buildsystem)

The devkitPro team broke the update script a few days ago which means three of our scripts are broken now -_-

In case somebody wants to continue working on this (I won't before vacation):

https://github.com/Ghabry/easyrpg-buildscripts/tree/pacman

3DS: sf2dlib doesn't compile anymore :/
Wii: Fails in libsndfile with
```
[...]/easyrpg-buildscripts/wii/opt/devkitpro/devkitPPC/lib/gcc/powerpc-eabi/8.1.0/include/ssp/stdio.h:38:10: fatal error: ssp.h: No such file or directory                          
 #include <ssp.h>
```
Switch: Compiles fine

--- a/3ds/1_download_library.sh
+++ b/3ds/1_download_library.sh
@@ -8,12 +8,12 @@ export WORKSPACE=$PWD
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source $SCRIPT_DIR/../shared/import.sh
 
-msg " [1] Installing devkitARM"
+msg " [1] Checking devkitARM"
 
-rm -rf $PWD/devkitPro
-download https://raw.githubusercontent.com/devkitPro/installer/master/perl/devkitARMupdate.pl
-perl -pi -e "s|/opt/devkitpro|$PWD/devkitPro|" devkitARMupdate.pl
-perl devkitARMupdate.pl
+if [[ -z $DEVKITPRO || -z $DEVKITARM ]]; then
+	echo "Setup devkitARM properly. \$DEVKITPRO and \$DEVKITARM need to be set."
+	exit 1
+fi
 
 msg " [2] Downloading generic libraries"
 
@@ -45,10 +45,6 @@ download_and_extract $EXPAT_URL
 rm -rf $LIBOGG_DIR
 download_and_extract $LIBOGG_URL
 
-# libvorbis
-rm -rf $LIBVORBIS_DIR
-download_and_extract $LIBVORBIS_URL
-
 # tremor
 rm -rf $TREMOR_DIR
 download $TREMOR_URL -O $TREMOR_FILE
@@ -74,7 +70,13 @@ download_and_extract $SPEEXDSP_URL
 rm -rf $WILDMIDI_DIR
 download_and_extract $WILDMIDI_URL
 
-# opus not supported, because of -ffast-math
+# opus
+rm -rf $OPUS_DIR
+download_and_extract $OPUS_URL
+
+# opusfile
+rm -rf $OPUSFILE_DIR
+download_and_extract $OPUSFILE_URL
 
 # ICU
 rm -rf $ICU_DIR
@@ -84,7 +86,3 @@ download_and_extract $ICU_URL
 rm -f $ICUDATA_FILES
 download_and_extract $ICUDATA_URL
 
-msg " [3] Downloading platform libraries"
-
-rm -rf sf2dlib
-git_clone https://github.com/xerpi/sf2dlib.git

--- a/3ds/2_build_toolchain.sh
+++ b/3ds/2_build_toolchain.sh
@@ -14,25 +14,31 @@ nproc=$(nproc)
 # Use ccache?
 test_ccache
 
+# Toolchain available?
+if [[ -z $DEVKITPRO || -z $DEVKITARM ]]; then
+	echo "Setup devkitARM properly. \$DEVKITPRO and \$DEVKITARM need to be set."
+	exit 1
+fi
+
 if [ ! -f .patches-applied ]; then
-	echo "patching libraries"
+	echo "Patching libraries"
 
 	patches_common
 
 	# Fix mpg123
-	pushd $MPG123_DIR
-	patch -Np1 < $SCRIPT_DIR/../shared/extra/mpg123.patch
-	autoreconf -fi
-	popd
+	(cd $MPG123_DIR
+		patch -Np1 < $SCRIPT_DIR/../shared/extra/mpg123.patch
+		autoreconf -fi
+	)
 
 	# Fix libsndfile
-	pushd $LIBSNDFILE_DIR
-	patch -Np1 < $SCRIPT_DIR/../shared/extra/libsndfile.patch
-	autoreconf -fi
-	popd
+	(cd $LIBSNDFILE_DIR
+		patch -Np1 < $SCRIPT_DIR/../shared/extra/libsndfile.patch
+		autoreconf -fi
+	)
 
-	# Disable pthread and other newlib issues
 	cp -rup icu icu-native
+	# Disable pthread and other newlib issues
 	patch -Np0 < $SCRIPT_DIR/icu59-3ds.patch
 
 	touch .patches-applied
@@ -42,8 +48,6 @@ cd $WORKSPACE
 
 echo "Preparing toolchain"
 
-export DEVKITPRO=$WORKSPACE/devkitPro
-export DEVKITARM=$DEVKITPRO/devkitARM
 export PATH=$DEVKITARM/bin:$PATH
 
 export PLATFORM_PREFIX=$WORKSPACE
@@ -59,19 +63,10 @@ function set_build_flags {
 		export CC="ccache $CC"
 		export CXX="ccache $CXX"
 	fi
-	export CFLAGS="-g0 -O2 -mword-relocations -fomit-frame-pointer -ffast-math -march=armv6k -mtune=mpcore -mfloat-abi=hard"
-	export CXXFLAGS=$CFLAGS
+	export CFLAGS="-g0 -O2 -mword-relocations -fomit-frame-pointer -ffunction-sections -march=armv6k -mtune=mpcore -mfloat-abi=hard -mtp=soft"
+	export CXXFLAGS="$CFLAGS -fno-exceptions"
 	export CPPFLAGS="-I$PLATFORM_PREFIX/include -D_3DS"
 	export LDFLAGS="-L$PLATFORM_PREFIX/lib"
-}
-
-function install_lib_sf2d() {
-	cd sf2dlib/libsf2d/
-	make clean
-	make
-	cp -r include/* ../../include/
-	cp -r lib/* ../../lib/
-	cd ../..
 }
 
 # Build native ICU
@@ -88,14 +83,12 @@ install_lib $FREETYPE_DIR $FREETYPE_ARGS --with-harfbuzz
 install_lib $PIXMAN_DIR $PIXMAN_ARGS
 install_lib_cmake $EXPAT_DIR $EXPAT_ARGS
 install_lib $LIBOGG_DIR $LIBOGG_ARGS
-install_lib $LIBVORBIS_DIR $LIBVORBIS_ARGS
 install_lib $TREMOR_DIR $TREMOR_ARGS
 install_lib $MPG123_DIR $MPG123_ARGS
 install_lib $LIBSNDFILE_DIR $LIBSNDFILE_ARGS
 install_lib_cmake $LIBXMP_LITE_DIR $LIBXMP_LITE_ARGS
 install_lib $SPEEXDSP_DIR $SPEEXDSP_ARGS
 install_lib_cmake $WILDMIDI_DIR $WILDMIDI_ARGS
+install_lib $OPUS_DIR $OPUS_ARGS
+install_lib $OPUSFILE_DIR $OPUSFILE_ARGS
 install_lib_icu_cross
-
-# Platform libs
-install_lib_sf2d

--- a/3ds/3_cleanup.sh
+++ b/3ds/3_cleanup.sh
@@ -8,6 +8,4 @@ source $SCRIPT_DIR/../shared/import.sh
 
 cleanup
 
-rm -rf sf2dlib/
-
 echo " -> done"

--- a/3ds/README.md
+++ b/3ds/README.md
@@ -2,11 +2,11 @@
 
 ## Nintendo 3DS Toolchain and libraries
 
-Local build process:
+### Prequisites:
 
-Run `0_build_everything.sh` in a terminal
+- Install devkitARM. This includes libctru, citro2d, citro3d and 3dstools.
+  (Environment variables `DEVKITPRO` and `DEVKITARM` need to be set)
 
-Usage:
+### Local build process:
 
-Set environment variable `DEVKITPRO` to `$PWD/devkitPro`
-and `DEVKITARM` to `$DEVKITPRO/devkitARM`
+- Run `0_build_everything.sh` in a terminal

--- a/switch/1_download_library.sh
+++ b/switch/1_download_library.sh
@@ -8,12 +8,12 @@ export WORKSPACE=$PWD
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source $SCRIPT_DIR/../shared/import.sh
 
-msg " [1] Installing devkitA64"
+msg " [1] Checking devkitA64"
 
-rm -rf devkitPro/
-wget -nv https://raw.githubusercontent.com/devkitPro/installer/master/perl/devkitA64update.pl
-perl -pi -e "s|/opt/devkitpro|$PWD/devkitPro|" devkitA64update.pl
-perl devkitA64update.pl
+if [[ -z $DEVKITPRO || ! -d "$DEVKITPRO/devkitA64" ]]; then
+	echo "Setup devkitA64 properly. \$DEVKITPRO needs to be set."
+	exit 1
+fi
 
 msg " [2] Downloading generic libraries"
 
@@ -44,10 +44,6 @@ download_and_extract $EXPAT_URL
 # libogg
 rm -rf $LIBOGG_DIR
 download_and_extract $LIBOGG_URL
-
-# libvorbis
-rm -rf $LIBVORBIS_DIR
-download_and_extract $LIBVORBIS_URL
 
 # tremor
 rm -rf $TREMOR_DIR
@@ -91,10 +87,3 @@ download_and_extract $ICU_URL
 rm -f $ICUDATA_FILES
 download_and_extract $ICUDATA_URL
 
-msg " [3] Downloading platform libraries"
-
-# Remove this later when libnx becomes more stable
-# (also shipped by devkitA64)
-
-rm -rf libnx
-git_clone https://github.com/switchbrew/libnx

--- a/switch/2_build_toolchain.sh
+++ b/switch/2_build_toolchain.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-# abort on errors
+# abort on error
 set -e
 
 export WORKSPACE=$PWD
@@ -14,38 +14,42 @@ nproc=$(nproc)
 # Use ccache?
 test_ccache
 
+# Toolchain available?
+if [[ -z $DEVKITPRO || ! -d "$DEVKITPRO/devkitA64" ]]; then
+	echo "Setup devkitA64 properly. \$DEVKITPRO needs to be set."
+	exit 1
+fi
+
 if [ ! -f .patches-applied ]; then
 	echo "Patching libraries"
 
 	patches_common
 
-	cp -rup icu icu-native
-
 	# Fix mpg123
-	pushd $MPG123_DIR
-	patch -Np1 < $SCRIPT_DIR/../shared/extra/mpg123.patch
-	autoreconf -fi
-	popd
+	(cd $MPG123_DIR
+		patch -Np1 < $SCRIPT_DIR/../shared/extra/mpg123.patch
+		autoreconf -fi
+	)
 
 	# Fix libsndfile
-	pushd $LIBSNDFILE_DIR
-	patch -Np1 < $SCRIPT_DIR/../shared/extra/libsndfile.patch
-	autoreconf -fi
-	popd
+	(cd $LIBSNDFILE_DIR
+		patch -Np1 < $SCRIPT_DIR/../shared/extra/libsndfile.patch
+		autoreconf -fi
+	)
 
 	# Wildmidi: Switch compatibility
-	pushd $WILDMIDI_DIR
-	patch -Np1 < $SCRIPT_DIR/wildmidi-switch.patch
-	popd
+	(cd $WILDMIDI_DIR
+		patch -Np1 < $SCRIPT_DIR/wildmidi-switch.patch
+	)
 
 	# disable libsamplerate examples and tests
-	pushd $LIBSAMPLERATE_DIR
-	perl -pi -e 's/examples tests//' Makefile.am
-	autoreconf -fi
-	popd
+	(cd $LIBSAMPLERATE_DIR
+		perl -pi -e 's/examples tests//' Makefile.am
+		autoreconf -fi
+	)
 
-	# Fix icu build
 	cp -rup icu icu-native
+	# Fix icu build
 	patch -Np0 < $SCRIPT_DIR/icu59-switch.patch
 
 	touch .patches-applied
@@ -55,13 +59,12 @@ cd $WORKSPACE
 
 echo "Preparing toolchain"
 
-export DEVKITPRO=${WORKSPACE}/devkitPro
-export PATH=${DEVKITPRO}/devkitA64/bin:$PATH
+export PATH=$DEVKITPRO/devkitA64/bin:$PATH
 
 export PLATFORM_PREFIX=$WORKSPACE
 export TARGET_HOST=aarch64-none-elf
-export PKG_CONFIG_PATH=$PLATFORM_PREFIX/lib/pkgconfig
-export PKG_CONFIG_LIBDIR=$PKG_CONFIG_PATH
+unset PKG_CONFIG_PATH
+export PKG_CONFIG_LIBDIR=$PLATFORM_PREFIX/lib/pkgconfig
 export MAKEFLAGS="-j${nproc:-2}"
 
 function set_build_flags {
@@ -71,17 +74,10 @@ function set_build_flags {
 		export CC="ccache $CC"
 		export CXX="ccache $CXX"
 	fi
-	export CFLAGS="-g0 -O2 -march=armv8-a -mtune=cortex-a57 -mtp=soft -fPIC -ftls-model=local-exec"
+	export CFLAGS="-g0 -O2 -march=armv8-a -mtune=cortex-a57 -mtp=soft -fPIC -ftls-model=local-exec -ffunction-sections"
 	export CXXFLAGS=$CFLAGS
-	export CPPFLAGS="-I$WORKSPACE/include -I$DEVKITPRO/libnx/include -DSWITCH"
-	export LDFLAGS="-L$WORKSPACE/lib"
-}
-
-function install_lib_nx {
-	cd libnx
-	make clean
-	make install
-	cd ..
+	export CPPFLAGS="-I$PLATFORM_PREFIX/include -I$DEVKITPRO/libnx/include -DSWITCH"
+	export LDFLAGS="-L$PLATFORM_PREFIX/lib"
 }
 
 # Build native icu59
@@ -89,8 +85,6 @@ install_lib_icu_native
 
 # Install libraries
 set_build_flags
-
-install_lib_nx
 
 install_lib_zlib
 install_lib $LIBPNG_DIR $LIBPNG_ARGS
@@ -100,7 +94,6 @@ install_lib $FREETYPE_DIR $FREETYPE_ARGS --with-harfbuzz
 install_lib $PIXMAN_DIR $PIXMAN_ARGS
 install_lib_cmake $EXPAT_DIR $EXPAT_ARGS
 install_lib $LIBOGG_DIR $LIBOGG_ARGS
-install_lib $LIBVORBIS_DIR $LIBVORBIS_ARGS
 install_lib $TREMOR_DIR $TREMOR_ARGS
 install_lib $MPG123_DIR $MPG123_ARGS
 install_lib $LIBSNDFILE_DIR $LIBSNDFILE_ARGS

--- a/switch/README.md
+++ b/switch/README.md
@@ -2,10 +2,12 @@
 
 ## Switch Toolchain and libraries
 
-Local build process:
+### Prequisites:
 
-Run `0_build_everything.sh` in a terminal
+- Install devkitA64. This includes libnx and switch-tools.
+  (Environment variables `DEVKITPRO` needs to be set)
 
-Usage:
+### Local build process:
 
-Set environment variable `DEVKITPRO` to `$PWD/devkitPro`.
+- Run `0_build_everything.sh` in a terminal
+

--- a/wii/1_download_library.sh
+++ b/wii/1_download_library.sh
@@ -8,7 +8,7 @@ export WORKSPACE=$PWD
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source $SCRIPT_DIR/../shared/import.sh
 
-# Override ICU version to 58.1
+# Override ICU version to 58.1, custom SDL
 source $SCRIPT_DIR/packages.sh
 
 msg " [1] Checking devkitPPC"
@@ -92,5 +92,5 @@ download_and_extract $ICUDATA_URL
 msg " [3] Downloading platform libraries"
 
 # SDL+SDL_mixer
-rm -rf sdl-wii/
-git_clone https://github.com/dborth/sdl-wii.git
+rm -rf $SDL_DIR
+download_and_extract $SDL_URL

--- a/wii/1_download_library.sh
+++ b/wii/1_download_library.sh
@@ -11,12 +11,12 @@ source $SCRIPT_DIR/../shared/import.sh
 # Override ICU version to 58.1
 source $SCRIPT_DIR/packages.sh
 
-msg " [1] Installing devkitPPC"
+msg " [1] Checking devkitPPC"
 
-rm -rf devkitPro/
-download https://raw.githubusercontent.com/devkitPro/installer/master/perl/devkitPPCupdate.pl
-perl -pi -e "s|/opt/devkitpro|$PWD/devkitPro|" devkitPPCupdate.pl
-perl devkitPPCupdate.pl
+if [[ -z $DEVKITPRO || -z $DEVKITPPC ]]; then
+	echo "Setup devkitPPC properly. \$DEVKITPRO and \$DEVKITPPC need to be set."
+	exit 1
+fi
 
 msg " [2] Downloading generic libraries"
 
@@ -47,10 +47,6 @@ download_and_extract $EXPAT_URL
 # libogg
 rm -rf $LIBOGG_DIR
 download_and_extract $LIBOGG_URL
-
-# libvorbis
-rm -rf $LIBVORBIS_DIR
-download_and_extract $LIBVORBIS_URL
 
 # tremor
 rm -rf $TREMOR_DIR

--- a/wii/2_build_toolchain.sh
+++ b/wii/2_build_toolchain.sh
@@ -17,22 +17,28 @@ nproc=$(nproc)
 # Use ccache?
 test_ccache
 
+# Toolchain available?
+if [[ -z $DEVKITPRO || -z $DEVKITPPC ]]; then
+	echo "Setup devkitPPC properly. \$DEVKITPRO and \$DEVKITPPC need to be set."
+	exit 1
+fi
+
 if [ ! -f .patches-applied ]; then
-	echo "patching libraries"
+	echo "Patching libraries"
 
 	patches_common
 
 	# Fix mpg123
-	pushd $MPG123_DIR
-	patch -Np1 < $SCRIPT_DIR/../shared/extra/mpg123.patch
-	autoreconf -fi
-	popd
+	(cd $MPG123_DIR
+		patch -Np1 < $SCRIPT_DIR/../shared/extra/mpg123.patch
+		autoreconf -fi
+	)
 
 	# Fix libsndfile
-	pushd $LIBSNDFILE_DIR
-	patch -Np1 < $SCRIPT_DIR/../shared/extra/libsndfile.patch
-	autoreconf -fi
-	popd
+	(cd $LIBSNDFILE_DIR
+		patch -Np1 < $SCRIPT_DIR/../shared/extra/libsndfile.patch
+		autoreconf -fi
+	)
 
 	cp -rup icu icu-native
 	# Fix ICU compilation problems on Wii
@@ -53,14 +59,12 @@ cd $WORKSPACE
 
 echo "Preparing toolchain"
 
-export DEVKITPRO=${WORKSPACE}/devkitPro
-export DEVKITPPC=${DEVKITPRO}/devkitPPC
 export PATH=$DEVKITPPC/bin:$PATH
 
 export PLATFORM_PREFIX=$WORKSPACE
 export TARGET_HOST=powerpc-eabi
-export PKG_CONFIG_PATH=$PLATFORM_PREFIX/lib/pkgconfig
-export PKG_CONFIG_LIBDIR=$PKG_CONFIG_PATH
+unset PKG_CONFIG_PATH
+export PKG_CONFIG_LIBDIR=$PLATFORM_PREFIX/lib/pkgconfig
 export MAKEFLAGS="-j${nproc:-2}"
 
 function set_build_flags {
@@ -70,7 +74,7 @@ function set_build_flags {
 		export CC="ccache $CC"
 		export CXX="ccache $CXX"
 	fi
-	export CFLAGS="-g -O2"
+	export CFLAGS="-g0 -O2 -mcpu=750 -meabi -mhard-float -ffunction-sections -fdata-sections"
 	export CXXFLAGS=$CFLAGS
 	export CPPFLAGS="-I$PLATFORM_PREFIX/include -DGEKKO"
 	export LDFLAGS="-L$PLATFORM_PREFIX/lib"
@@ -116,7 +120,6 @@ install_lib $FREETYPE_DIR $FREETYPE_ARGS --with-harfbuzz
 install_lib $PIXMAN_DIR $PIXMAN_ARGS --disable-vmx
 install_lib_cmake $EXPAT_DIR $EXPAT_ARGS
 install_lib $LIBOGG_DIR $LIBOGG_ARGS
-install_lib $LIBVORBIS_DIR $LIBVORBIS_ARGS
 install_lib $TREMOR_DIR $TREMOR_ARGS
 install_lib $MPG123_DIR $MPG123_ARGS
 install_lib $LIBSNDFILE_DIR $LIBSNDFILE_ARGS

--- a/wii/2_build_toolchain.sh
+++ b/wii/2_build_toolchain.sh
@@ -8,7 +8,7 @@ export WORKSPACE=$PWD
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source $SCRIPT_DIR/../shared/import.sh
 
-# Override ICU version to 58.1
+# Override ICU version to 58.1, custom SDL
 source $SCRIPT_DIR/packages.sh
 
 # Number of CPU
@@ -37,6 +37,8 @@ if [ ! -f .patches-applied ]; then
 	# Fix libsndfile
 	(cd $LIBSNDFILE_DIR
 		patch -Np1 < $SCRIPT_DIR/../shared/extra/libsndfile.patch
+		# do not fortify source
+		perl -pi -e 's/AX_ADD_FORTIFY_SOURCE//' configure.ac
 		autoreconf -fi
 	)
 
@@ -47,10 +49,7 @@ if [ ! -f .patches-applied ]; then
 	patch -Np0 < icu-pkg_genc.patch
 
 	# Patch SDL+SDL_mixer
-	cd sdl-wii
-	git reset --hard
-	cd ..
-	patch --binary -Np0 < $SCRIPT_DIR/sdl-wii.patch
+	patch -d $SDL_DIR --binary -Np1 < $SCRIPT_DIR/sdl-wii.patch
 
 	touch .patches-applied
 fi
@@ -85,7 +84,7 @@ function install_lib_sdl() {
 	echo "**** Building SDL ****"
 	echo ""
 
-	cd sdl-wii/SDL
+	cd $SDL_DIR/SDL
 	make clean
 	make install INSTALL_HEADER_DIR="$WORKSPACE/include" INSTALL_LIB_DIR="$WORKSPACE/lib"
 	cd ../..
@@ -98,7 +97,7 @@ function install_lib_sdlmixer() {
 	echo "**** Building SDL_mixer ****"
 	echo ""
 
-	cd sdl-wii/SDL_mixer
+	cd $SDL_DIR/SDL_mixer
 	make clean
 	make install INSTALL_HEADER_DIR="$WORKSPACE/include" INSTALL_LIB_DIR="$WORKSPACE/lib"
 	cd ../..

--- a/wii/3_cleanup.sh
+++ b/wii/3_cleanup.sh
@@ -8,6 +8,6 @@ source $SCRIPT_DIR/../shared/import.sh
 
 cleanup
 
-rm -rf sdl-wii/
+rm -rf sdl-wii-*/
 
 echo " -> done"

--- a/wii/README.md
+++ b/wii/README.md
@@ -2,11 +2,12 @@
 
 ## Wii Toolchain and libraries
 
-Local build process:
+### Prequisites:
 
-Run `0_build_everything.sh` in a terminal
+- Install devkitPPC. This includes libogc, libfat and gamecube-tools.
+  (Environment variables `DEVKITPRO` and `DEVKITPPC` need to be set)
 
-Usage:
+### Local build process:
 
-Set environment variable `DEVKITPRO` to `$PWD/devkitPro`
-and `DEVKITPPC` to `$DEVKITPRO/devkitPPC`
+- Run `0_build_everything.sh` in a terminal
+

--- a/wii/icu-wii.patch
+++ b/wii/icu-wii.patch
@@ -133,8 +133,8 @@ diff -Naur icu-native/source/common/umutex.h icu/source/common/umutex.h
 +
 +#define PTHREAD_MUTEX_INITIALIZER 0
 +#define PTHREAD_COND_INITIALIZER 0
-+typedef int pthread_mutex_t;
-+typedef int pthread_cond_t;
++typedef uint32_t pthread_mutex_t;
++typedef uint32_t pthread_cond_t;
  
  struct UMutex {
      pthread_mutex_t  fMutex;

--- a/wii/packages.sh
+++ b/wii/packages.sh
@@ -1,3 +1,8 @@
 lib=ICU
 ver=58.1
 ICU_URL=http://download.icu-project.org/files/icu4c/$ver/icu4c-${ver/./_}-src.tgz
+
+lib=sdl-wii
+ver=2018-08-31
+SDL_URL="https://github.com/dborth/$lib/archive/$ver.tar.gz"
+SDL_DIR="$lib-$ver"

--- a/wii/sdl-wii.patch
+++ b/wii/sdl-wii.patch
@@ -1,6 +1,6 @@
-diff -Naur sdl-wii-orig/SDL/Makefile sdl-wii/SDL/Makefile
---- sdl-wii-orig/SDL/Makefile	2016-08-20 12:20:23.723226343 +0200
-+++ sdl-wii/SDL/Makefile	2016-08-20 10:54:29.209385683 +0200
+diff --color -Naur sdl-wii.orig/SDL/Makefile sdl-wii-2018-08-31/SDL/Makefile
+--- sdl-wii.orig/SDL/Makefile	2018-08-31 18:49:59.000000000 +0200
++++ sdl-wii-2018-08-31/SDL/Makefile	2018-10-13 13:33:56.768210370 +0200
 @@ -87,7 +87,7 @@
  	@echo ----
  
@@ -10,9 +10,9 @@ diff -Naur sdl-wii-orig/SDL/Makefile sdl-wii/SDL/Makefile
  INCLUDES		:= -Iinclude -I$(LIBOGC_INC)
  DEFINES			:= -DGEKKO
  CFLAGS			:= $(COMMON_FLAGS) $(INCLUDES) $(DEFINES)
-diff -Naur sdl-wii-orig/SDL/src/video/wii/SDL_wiivideo.c sdl-wii/SDL/src/video/wii/SDL_wiivideo.c
---- sdl-wii-orig/SDL/src/video/wii/SDL_wiivideo.c	2016-08-20 12:20:23.829894660 +0200
-+++ sdl-wii/SDL/src/video/wii/SDL_wiivideo.c	2016-08-20 10:54:29.209385683 +0200
+diff --color -Naur sdl-wii.orig/SDL/src/video/wii/SDL_wiivideo.c sdl-wii-2018-08-31/SDL/src/video/wii/SDL_wiivideo.c
+--- sdl-wii.orig/SDL/src/video/wii/SDL_wiivideo.c	2018-08-31 18:49:59.000000000 +0200
++++ sdl-wii-2018-08-31/SDL/src/video/wii/SDL_wiivideo.c	2018-10-13 13:33:56.768210370 +0200
 @@ -282,7 +282,7 @@
  	mode_320.h = mode_640.h / 2;
  
@@ -22,9 +22,9 @@ diff -Naur sdl-wii-orig/SDL/src/video/wii/SDL_wiivideo.c sdl-wii/SDL/src/video/w
  	vformat->BytesPerPixel	= 2;
  
  	this->hidden->buffer = NULL;
-diff -Naur sdl-wii-orig/SDL_mixer/Makefile sdl-wii/SDL_mixer/Makefile
---- sdl-wii-orig/SDL_mixer/Makefile	2016-08-20 12:20:23.889895589 +0200
-+++ sdl-wii/SDL_mixer/Makefile	2016-08-20 11:28:52.160536313 +0200
+diff --color -Naur sdl-wii.orig/SDL_mixer/Makefile sdl-wii-2018-08-31/SDL_mixer/Makefile
+--- sdl-wii.orig/SDL_mixer/Makefile	2018-08-31 18:49:59.000000000 +0200
++++ sdl-wii-2018-08-31/SDL_mixer/Makefile	2018-10-13 13:36:49.461845562 +0200
 @@ -24,7 +24,7 @@
  PIPE_TO_SED := 2>&1 | sed "s/:\([0-9]*\):/\(\1\) :/"
  
@@ -39,9 +39,9 @@ diff -Naur sdl-wii-orig/SDL_mixer/Makefile sdl-wii/SDL_mixer/Makefile
  
  # Compilation flags.
 -COMMON_FLAGS	:= -g -O3 -mrvl -Wall $(MACHDEP)
--INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(LIBOGC_INC)/SDL -I$(LIBOGC_INC)/mpeg -I$(DEVKITPRO)/libsmpeg/include -Isrc/mikmod -Isrc/timidity
+-INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(LIBOGC_INC)/SDL -I$(LIBOGC_INC)/mpeg -I../smpeg/include -Isrc/mikmod -Isrc/timidity -I../SDL/include
 +COMMON_FLAGS	:= -g -O2 -mrvl -Wall $(MACHDEP)
-+INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(WORKSPACE)/include -I$(WORKSPACE)/include/SDL
++INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I../SDL/include
  # -include sys/wait.h 
  # -include sys/syslimits.h
  # -Isrc/native_midi

